libmapi/oc_log.c

```diff
@@ -0,0 +1,71 @@
+/*
+   Logging API
+
+   OpenChange Project
+
+   Copyright (C) Jelmer Vernooij 2015
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libmapi/libmapi.h"
+
+void oc_log(enum oc_log_level level, const char *fmt_string, ...)
+{
+	va_list ap;
+	va_start(ap, fmt_string);
+	oc_logv(level, fmt_string, ap);
+	va_end(ap);
+}
+
+void oc_logv(enum oc_log_level level, const char *fmt_string, va_list ap)
+{
+	char line[OC_LOG_MAX_LINE];
+	int samba_level;
+	int nwritten;
+
+	nwritten = vsnprintf(line, sizeof(line), fmt_string, ap);
+
+	if (level <= 0) {
+		samba_level = -level;
+	} else {
+		samba_level = 0;
+	}
+
+	DEBUG(samba_level, ("%s\n", line));
```

Comment:
I'd like to let the developer to insert the new line or not in the `fmt_string` as `DEBUG` macro from samba does.